src/backend/distributed/sql/citus--9.5-1--10.0-1.sql

```diff
@@ -7,3 +7,29 @@ DROP FUNCTION IF EXISTS pg_catalog.citus_total_relation_size(regclass);
 #include "udfs/citus_finish_pg_upgrade/10.0-1.sql"
 
 #include "../../columnar/sql/columnar--9.5-1--10.0-1.sql"
+
+-- Remove all the appended shardId from the names of constraints of partitioned tables,
+-- with the eception of foreign key constraints that should keep the shardId in their names
+DO $$DECLARE r record;
+BEGIN
+    FOR r IN SELECT cls.relname::text table_name,
+					cons.conname prev_constraint_name,
+                    constraint_name_and_shardid[1] as new_constraint_name
+                FROM pg_catalog.pg_constraint cons
+                    JOIN pg_class cls ON cons.conrelid = cls.oid,
+                    -- capture the original constraint name and the shardid in seperate groups
+                    regexp_match(cons.conname, '^(.*)_([0-9]+)$') as constraint_name_and_shardid,
+                    -- capture the parent table name and the shardid in seperate groups
+                    regexp_match(cls.relname::text, '^(.*)_([0-9]+)$') as parent_name_and_shardid
+                WHERE
+                    -- ignore foreign key constraints
+                    cons.contype <> 'f'
+                    -- find only partitioned tables
+                    AND cls.relkind = 'p'
+                    -- check that the shardid in the constraint name and the parent table are the same
+                    AND parent_name_and_shardid[2] = constraint_name_and_shardid[2]
```

Comment:
This script seems a bit dangerous to me, For example what happens if my constraint name has a suffix that is like a shard id, as in what if the original name is: `my_check_2131321_121112` and when users upgrade it will modify the original name I guess?

Reply:
It will rename the constraint only if the shardId of the parent table is `121112`.

Reply:
Removed the migration file completely.